src/commands/play.ts

```diff
@@ -63,38 +59,74 @@ export default class Play extends Command {
 				return;
 			}
 
-			const [url] = params;
-			const id = ytdl.getURLVideoID(url);
+			let song: Omit<Song, "requestingUserId">;
 
-			const { items } = await fetchVideoMeta(id);
+			if (ytdl.validateURL(params[0]))
+				song = await this.getSongDetailsByUrl(params[0]);
+			else song = await this.getSongDetailsByQuery(params.join(" "));
```

Comment:
This could be abstracted by using Strategy Pattern where the strategy is getSongDetails, regardless implementation.